client/my-sites/checkout/composite-checkout/lib/submit-wpcom-transaction.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Internal dependencies
+ */
+import type { PaymentProcessorOptions } from '../types/payment-processors';
+import { createAccount } from '../payment-method-helpers';
+import wp from 'calypso/lib/wp';
+import type {
+	WPCOMTransactionEndpointRequestPayload,
+	WPCOMTransactionEndpointResponse,
+} from '../types/transaction-endpoint';
+
+export default async function submitWpcomTransaction(
+	payload: WPCOMTransactionEndpointRequestPayload,
+	transactionOptions: PaymentProcessorOptions
+): Promise< WPCOMTransactionEndpointResponse > {
+	if ( transactionOptions && transactionOptions.createUserAndSiteBeforeTransaction ) {
+		return createAccount().then( ( response ) => {
+			const siteIdFromResponse = response?.blog_details?.blogid;
+
+			// If the account is already created(as happens when we are reprocessing after a transaction error), then
+			// the create account response will not have a site ID, so we fetch from state.
+			const siteId = siteIdFromResponse || transactionOptions.siteId;
```

Comment:
Per the testing instructions, just noting that here we're getting the site ID from `transactionOptions` rather than `select( 'wpcom' )?.getSiteId()`. I see where that property is added to the `PaymentProcessorOptions` type in this PR, but I'm not sure exactly where that happens. I was able to complete a purchase so the site ID is getting set somewhere.

Reply:
Good point. It is being added to `transactionOptions` here: https://github.com/Automattic/wp-calypso/blob/a6c3e5b44d8629ffa39c3d2930e679be49f7fd00/client/my-sites/checkout/composite-checkout/composite-checkout.tsx#L442 so it should be safe (I'd like to deprecate `getSiteId` on the wpcom store eventually as it's a global crutch). Thanks for noticing that though!